main/test/flix/Test.Type.Bool.flix

```diff
@@ -447,26 +616,71 @@ mod Test.Type.Bool {
       ((b1 and (not b2)) or (b2 and (not b1)))] = I2.I2
-    def mul(_: I2[a1, b1], _: I2[a2, b2]):
+    def mul2(_: I2[a1, b1], _: I2[a2, b2]):
       I2[(((a1 and b2) and (not (b1 and a2))) or ((b1 and a2) and (not (a1 and b2)))),
       (b1 and b2)] = I2.I2
 
+    def mkZero4(): I4[false, false, false, false] = I4.I4
+    def mkOne4(): I4[false, false, false, true] = I4.I4
+    def mkTwo4(): I4[false, false, true, false] = I4.I4
+    def mkThree4(): I4[false, false, true, true] = I4.I4
+    def mkFour4(): I4[false, true, false, false] = I4.I4
+    def mkFive4(): I4[false, true, false, true] = I4.I4
+    def mkSix4(): I4[false, true, true, false] = I4.I4
+    def mkSeven4(): I4[false, true, true, true] = I4.I4
+    def mkEight4(): I4[true, false, false, false] = I4.I4
+    def mkNine4(): I4[true, false, false, true] = I4.I4
+    def mkTen4(): I4[true, false, true, false] = I4.I4
+    def mkEleven4(): I4[true, false, true, true] = I4.I4
+    def mkTwelve4(): I4[true, true, false, false] = I4.I4
+    def mkThirteen4(): I4[true, true, false, true] = I4.I4
+    def mkFourteen4(): I4[true, true, true, false] = I4.I4
+    def mkFifteen4(): I4[true, true, true, true] = I4.I4
+
+    def isZero4(_: I4[false, false, false, false]): Bool = true
+    def isOne4(_: I4[false, false, false, true]): Bool = true
+    def isTwo4(_: I4[false, false, true, false]): Bool = true
+    def isThree4(_: I4[false, false, true, true]): Bool = true
+    def isFour4(_: I4[false, true, false, false]): Bool = true
+    def isFive4(_: I4[false, true, false, true]): Bool = true
+    def isSix4(_: I4[false, true, true, false]): Bool = true
+    def isSeven4(_: I4[false, true, true, true]): Bool = true
+    def isEight4(_: I4[true, false, false, false]): Bool = true
+    def isNine4(_: I4[true, false, false, true]): Bool = true
+    def isTen4(_: I4[true, false, true, false]): Bool = true
+    def isEleven4(_: I4[true, false, true, true]): Bool = true
+    def isTwelve4(_: I4[true, true, false, false]): Bool = true
+    def isThirteen4(_: I4[true, true, false, true]): Bool = true
+    def isFourteen4(_: I4[true, true, true, false]): Bool = true
+    def isFifteen4(_: I4[true, true, true, true]): Bool = true
+
+    def _bitwiseNot4(_: I4[a, b, c, d]): I4[not a, not b, not c, not d] = I4.I4
+    def _bitwiseOr4(_: I4[a1, b1, c1, d1], _: I4[a2, b2, c2, d2]):
+      I4[a1 or a2, b1 or b2, c1 or c2, d1 or d2] = I4.I4
+    def _bitwiseAnd4(_: I4[a1, b1, c1, d1], _: I4[a2, b2, c2, d2]):
+      I4[a1 and a2, b1 and b2, c1 and c2, d1 and d2] = I4.I4
+    def inc4(_: I4[a, b, c, d]): I4[a xor (b and c and d), b xor (c and d), c xor d, not d] = I4.I4
+    def add4(_: I4[a1, b1, c1, d1], _: I4[a2, b2, c2, d2]):
+      I4[(a1 xor a2 xor ((b1 and b2) or ((b1 xor b2) and ((c1 and c2) or ((c1 xor c2) and (d1 and d2)))))),
+        (b1 xor b2 xor ((c1 and c2) or ((c1 xor c2) and (d1 and d2)))),
```

Comment:
If you add one more space of indent to these lines this becomes easier to read :)